inst/templates/ad_adrs.R

```diff
@@ -172,23 +178,28 @@ adrs <- adrs %>%
       PARCAT1 = "Tumor Response",
       PARCAT2 = "Investigator",
       PARCAT3 = "Recist 1.1",
+      AVAL = aval_resp(AVALC),
       ANL01FL = "Y"
     )
   )
 
 # Best overall response of CR/PR
 adrs <- adrs %>%
-  derive_param_extreme_event(
-    dataset_adsl = adsl,
-    dataset_source = adrs,
-    filter_source = PARAMCD == "BOR" & AVALC %in% c("CR", "PR") & ANL01FL == "Y",
+  derive_extreme_records(
+    dataset_ref = adsl,
+    dataset_add = adrs,
+    by_vars = exprs(STUDYID, USUBJID),
+    filter_add = PARAMCD == "BOR" & AVALC %in% c("CR", "PR") & ANL01FL == "Y",
     order = exprs(ADT, RSSEQ),
+    mode = "first",
+    exist_flag = AVALC,
     set_values_to = exprs(
       PARAMCD = "BCP",
       PARAM = "Best Overall Response of CR/PR by Investigator (confirmation not required)",
       PARCAT1 = "Tumor Response",
       PARCAT2 = "Investigator",
       PARCAT3 = "Recist 1.1",
+      AVAL = yn_to_numeric(AVALC),
```

Comment:
Yes, I noticed this as well. Most likely it is due to the missed update of the PD derivation. In admiralonco 0.3.0 `AVALC` is set to `"PD"` but later when we filter on observations up to PD we expect `AVALC == "Y"`.

So I think the code is correct now but the test data should be improved. We need to move this to next release because admiral.test is going to be released today or tomorrow. There is also an ongoing discussion regarding reimplementing/restructuring admiral.test.

Reply:
I have created an issue for the test data update: https://github.com/pharmaverse/admiralonco/issues/226